open-metadata-implementation/access-services/asset-catalog/asset-catalog-spring/src/main/java/org/odpi/openmetadata/accessservices/assetcatalog/server/spring/AssetCatalogEntityResource.java

```diff
@@ -136,6 +136,39 @@ public AssetListResponse searchByType(@PathVariable("serverName") String serverN
         return assetService.searchByType(serverName, userId, searchCriteria, searchParameters);
     }
 
+    /**
+     * Return a list of assets by asset type name without any additional search criteria
+     *
+     * @param serverName       unique identifier for requested server.
+     * @param userId           the unique identifier for the user
+     * @param typeName         the assets type name to search for
+     * @return                 list of assets by type name
+     */
+    @GetMapping(path = "/assets-by-type-name/{typeName}",
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    public AssetListResponse getAssetsByTypeName(@PathVariable("serverName") String serverName,
+                                                 @PathVariable("userId") String userId,
+                                                 @PathVariable("typeName") @NotBlank String typeName) {
+        return assetService.searchByTypeNameOrGUID(serverName, userId, typeName, null);
+    }
+
+    /**
+     * Return a list of assets by asset type GUID without any additional search criteria
```

Comment:
I think that we need to be explicit in the comments here as to why we are expecting a list of assets when we are looking for a type by name or guid, when there is only one. I assume the reason for this is that it includes subtypes , if so we should say that in the comments / javadoc

Reply:
I'll include subtypes in the description